cmake/FindQuickLintJsEmacs.cmake

```
@@ -0,0 +1,81 @@
+# Copyright (C) 2020  Matthew "strager" Glazar
+# See end of file for extended copyright information.
+
+if (QUICK_LINT_JS_EMACS)
+  set(QUICK_LINT_JS_EMACS "${QUICK_LINT_JS_EMACS}")
+else ()
+  find_program(QUICK_LINT_JS_EMACS "emacs")
+endif ()
+
+if (NOT QUICK_LINT_JS_EMACS)
+  message(WARNING "Emacs not found. Skipping... ")
+  return ()
+endif ()
+
+execute_process(
+  COMMAND
+    ${QUICK_LINT_JS_EMACS}
+    -Q -batch
+    --eval "(princ (format \"%s.%s\" emacs-major-version emacs-minor-version))"
+  RESULT_VARIABLE EMACS_EXIT_CODE
+  OUTPUT_VARIABLE EMACS_VERSION)
+
+if (NOT EMACS_EXIT_CODE EQUAL 0)
+  message(WARNING "Emacs (${QUICK_LINT_JS_EMACS}) found but can't get its version. Skipping...")
+  return ()
+endif()
+
+if (NOT EMACS_VERSION GREATER_EQUAL 24.5)
+  message(WARNING "Emacs found (${QUICK_LINT_JS_EMACS}), but version ${EMACS_VERSION} is not supported. Skipping...")
+  return ()
+endif ()
+
+set(QUICK_LINT_JS_EMACS_FOUND TRUE)
+message(STATUS "Found Emacs: (${QUICK_LINT_JS_EMACS}) suitable version ${EMACS_VERSION} minimum required is 24.5")
```

Nit: Match format of similar messages in CMake:
```suggestion
message(STATUS "Found Emacs: ${QUICK_LINT_JS_EMACS} (found suitable version \"${EMACS_VERSION}\", minimum required is \"24.5\"")
```